Job 不支持事务注解 @Transactional

在一个继承了 SimpleJob 的任务内，在某个函数前加上 `@Transactional`注解后，启动时会报如下异常：
```
Caused by: com.dangdang.ddframe.job.exception.JobConfigurationException: Job conflict with register center. The job 'bill_day' in register center's class is 'xxx.task.BillJob', your job class is 'xxx.task.BillJob$$EnhancerBySpringCGLIB$$c6d64d7f'
```
报错在 ConfigurationService.class 的 79 行。看起来是检查 Zookeeper 配置的 config 中配置的 class
 名时抛出的异常。ZK 上 config 配置的的 class 为之前配置的 BillJob 类的全限定名。后来加上了 `@Transactional`之后，检验的类名成了 Spring 代理类的名称，于是就错了。

又试了把 Zookeeper 上该任务的 config 删掉，第一次启动成功了，第二次启动又报错：
```
Caused by: com.dangdang.ddframe.job.exception.JobConfigurationException: Job conflict with register center. The job 'bill_day' in register center's class is 'xxx.task.BillJob$$EnhancerBySpringCGLIB$$d030d726', your job class is 'xxx.task.BillJob$$EnhancerBySpringCGLIB$$ec4064c6'
```
意料之中。

这种问题该如何解决呢？难道 Job 类的代码里不能用事务么。又或者当我的 job 文件不能改名或不能路径不能变动么（会出现同样的类名校验问题）？

--- a/elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/schedule/JobScheduler.java
+++ b/elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/schedule/JobScheduler.java
@@ -26,6 +26,7 @@ import org.apache.shardingsphere.elasticjob.infra.exception.JobSystemException;
 import org.apache.shardingsphere.elasticjob.infra.handler.sharding.JobInstance;
 import org.apache.shardingsphere.elasticjob.lite.internal.guarantee.GuaranteeService;
 import org.apache.shardingsphere.elasticjob.lite.internal.setup.SetUpFacade;
+import org.apache.shardingsphere.elasticjob.lite.internal.util.ProxyUtils;
 import org.apache.shardingsphere.elasticjob.reg.base.CoordinatorRegistryCenter;
 import org.apache.shardingsphere.elasticjob.tracing.api.TracingConfiguration;
 import org.quartz.JobBuilder;
@@ -89,7 +90,8 @@ public final class JobScheduler {
         this.tracingConfig = tracingConfig;
         setUpFacade = new SetUpFacade(regCenter, jobConfig.getJobName(), this.elasticJobListeners);
         schedulerFacade = new SchedulerFacade(regCenter, jobConfig.getJobName());
-        this.jobConfig = setUpFacade.setUpJobConfiguration(elasticJob.getClass().getName(), jobConfig);
+        String jobClassName = ProxyUtils.isCglibProxy(elasticJob) ? ProxyUtils.getTargetClass(elasticJob).getName() : elasticJob.getClass().getName();
+        this.jobConfig = setUpFacade.setUpJobConfiguration(jobClassName, jobConfig);
         setGuaranteeServiceForElasticJobListeners(regCenter, this.elasticJobListeners);
         jobScheduleController = createJobScheduleController();
     }

--- a/elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/util/ProxyUtils.java
+++ b/elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/util/ProxyUtils.java
@@ -1,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.elasticjob.lite.internal.util;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+/**
+ * Aop target Utility.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class ProxyUtils {
+    
+    /**
+     * Get target class.
+     * 
+     * @param proxy proxy object
+     * @return target class
+     */
+    public static Class<?> getTargetClass(final Object proxy) {
+        if (!isCglibProxy(proxy)) {
+            return proxy.getClass();
+        }
+        return proxy.getClass().getSuperclass();
+    }
+
+    /**
+     * Check whether the given object is a CGLIB proxy.
+     */
+    public static boolean isCglibProxy(final Object object) {
+        return object.getClass().getName().contains("$$EnhancerBySpringCGLIB");
+    }
+}

--- a/elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/util/ProxyUtils.java
+++ b/elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/util/ProxyUtils.java
@@ -41,6 +41,8 @@ public final class ProxyUtils {
 
     /**
      * Check whether the given object is a CGLIB proxy.
+     * @param object job object
+     * @return if the object is cglib proxy
      */
     public static boolean isCglibProxy(final Object object) {
         return object.getClass().getName().contains("$$EnhancerBySpringCGLIB");

